app/Console/AutoImports.php

```diff
@@ -112,75 +120,65 @@ private function getExtension(string $file): string
         return strtolower($parts[count($parts) - 1]);
     }
 
-    /**
-     * This method only works on files with an extension with exactly three letters
-     * (i.e. "csv", "xml").
-     */
-    private function hasJsonConfiguration(string $directory, string $file): bool
+    private function getExtensionLength(string $file): int
     {
-        $short    = substr($file, 0, -4);
+        $parts = explode('.', $file);
+        if (1 === count($parts)) {
+            return 0;
+        }
+        return strlen($parts[count($parts) - 1]) + 1;
+    }
+
+    private function getJsonConfiguration(string $directory, string $file): ?string
+    {
+        $extensionLength = $this->getExtensionLength($file);
+        $short    = substr($file, 0, -$extensionLength);
         $jsonFile = sprintf('%s.json', $short);
         $fullJson = sprintf('%s/%s', $directory, $jsonFile);
 
-        if (!file_exists($fullJson)) {
-            $hasFallbackConfig = $this->hasFallbackConfig($directory);
-            if ($hasFallbackConfig) {
+        if (file_exists($fullJson)) {
```

Comment:
I think you can do this entire routine without `elseif` and `else` since you're returning things anyway.